workspace/build_manifest.go

```diff
@@ -59,41 +58,53 @@ type ExecPhase struct {
 
 func (e *ExecPhase) EnvironmentVariables(ctx context.Context, envName string, data runtime.RuntimeEnvironmentData) []string {
 
-	result := make([]string, 0)
+	envMap := make(map[string]string)
 
 	for _, property := range e.Environment["default"] {
 
 		if _, _, ok := plumbing.SaneEnvironmentVar(property); ok {
 			interpolated, err := TemplateToString(property, data)
 			if err == nil {
-				result = append(result, interpolated)
-			} else {
-				result = append(result, property)
+				if key, value, ok := plumbing.SaneEnvironmentVar(interpolated); ok {
+					envMap[key] = value
+				}
+			} else if key, value, ok := plumbing.SaneEnvironmentVar(property); ok {
+				envMap[key] = value
 			}
 		}
 	}
 
 	for k, v := range data.Containers.Environment(ctx) {
-		result = append(result, k, v)
+		envMap[k] = v
 	}
 
 	if envName != "default" {
 		for _, property := range e.Environment[envName] {
 			if _, _, ok := plumbing.SaneEnvironmentVar(property); ok {
-				result = append(result, property)
+				if key, value, ok := plumbing.SaneEnvironmentVar(property); ok {
+					envMap[key] = value
+				}
 			}
 		}
 	}
 
 	// Check for local .env file
```

Comment:
Can you help me understand what this previous code didn't do?

Reply:
Oh, it didn't override an already set environment variable, so we ended up with duplicated vars, each with different values. This is breaking some of our own projects, in some build targets (e.g: the API). I'll improve comments, and probably improve the logic itself, too many loops.